fbw-common/src/wasm/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -154,39 +242,75 @@ impl<const ZONES: usize, const ENGINES: usize> AirConditioningSystemController<Z
             .find_map(|&adiru_number| adirs.ground_speed(adiru_number).normal_value())
     }
 
-    pub fn pack_fault_determination(&self, pneumatic: &impl PackFlowValveState) -> [bool; 2] {
-        [
-            self.pack_flow_controller[Pack(1).to_index()].fcv_status_determination(pneumatic),
-            self.pack_flow_controller[Pack(2).to_index()].fcv_status_determination(pneumatic),
-        ]
-    }
-
-    pub(super) fn trim_air_valve_controllers(&self, zone_id: usize) -> TrimAirValveController {
-        self.trim_air_system_controller
-            .trim_air_valve_controllers(zone_id)
+    pub fn pack_fault_determination(&self, pneumatic: &impl PackFlowValveState) -> bool {
+        self.pack_flow_controller
+            .fcv_status_determination(pneumatic)
+            || self.both_channels_failure()
     }
 
     pub fn cabin_fans_controller(&self) -> CabinFanController<ZONES> {
         self.cabin_fans_controller
     }
 
-    pub fn individual_pack_flow(&self, pack_id: Pack) -> MassRate {
-        self.pack_flow_controller[pack_id.to_index()].pack_flow()
+    pub fn individual_pack_flow(&self) -> MassRate {
+        self.pack_flow_controller.pack_flow()
     }
 
     pub fn duct_demand_temperature(&self) -> Vec<ThermodynamicTemperature> {
-        self.zone_controller
+        let mut demand_temperature: Vec<ThermodynamicTemperature> = self
+            .zone_controller
             .iter()
             .map(|zone| zone.duct_demand_temperature())
-            .collect()
+            .collect();
+        // Because each ACSC calculates the demand of its respective zone(s), we fill the vector for the trim air system
+        let mut filler_vector = vec![
+            ThermodynamicTemperature::new::<degree_celsius>(24.);
+            ZONES - demand_temperature.len()
+        ];
+        if matches!(self.id, AcscId::Acsc1(_)) {
+            demand_temperature.extend(filler_vector);
+            demand_temperature
+        } else {
+            filler_vector.extend(demand_temperature);
+            filler_vector
+        }
```

Comment:
ACSC1 calculates the demand temperature for the cockpit [zone_id = 0] and ACSC2 for the cabin zones [zone_id = 1 & 2].

To keep this generic, what I did was "fill" the rest of the vector with dummy data in each ACSC so both would output a vector of the same size (so for ACSC1 [real data, dummy, dummy], for ACSC2 [dummy, real data, real data]. These vectors are then passed to the trim air system controller in each ACSC, but only the correct controllers (0 for ACSC1, and 1&2 for ACSC2) are passed to the trim air system to actuate the valves.

I know a bit confusing but it was the best way I found to do this and keep it generic 😄.

Reply:
Ok then I have the following suggestion how this function can be implemented with reducing the amount of allocations:
```rust
    pub fn duct_demand_temperature(&self) -> [ThermodynamicTemperature; ZONES] {
        let demand_temperature: Vec<ThermodynamicTemperature> = self
            .zone_controller
            .iter()
            .map(|zone| zone.duct_demand_temperature())
            .collect();
        // Because each ACSC calculates the demand of its respective zone(s), we fill the vector for the trim air system
        let mut filler_vector = [ThermodynamicTemperature::new::<degree_celsius>(24.); ZONES];
        if matches!(self.id, AcscId::Acsc1(_)) {
            filler_vector[..demand_temperature.len()].copy_from_slice(&demand_temperature);
        } else {
            filler_vector[demand_temperature.len()..].copy_from_slice(&demand_temperature);
        };
        filler_vector
    }
```

Reply:
Thanks! I like your suggestion but it doesn't seem to work as is. I've implemented something similar, as we know the cockpit will always be zone_id = 1.